lollipop/types.py

```diff
@@ -1292,3 +1293,42 @@ def __repr__(self):
             klass=self.__class__.__name__,
             inner_type=repr(self.inner_type),
         )
+
+
+class Transform(Type):
+    """
+
+    Example:
+
+        Point = namedtuple('Point', ['x', 'y'])
+
+        PointType = Transform(
+          Tuple(Integer(), Integer()),
+          load=lambda values: Point(values[0], values[1]),
+          dump=lambda point: [point.x, point.y],
+        )
```

Comment:
Please add example on how serialized/deserialized data will look like (since in this case it is not obvious)